### PR TITLE
add types for cucumberOpts, fix cucumberOpts in wdio-cli template

### LINKED ISSUE #4215: WebDriverIO v5 cucumberOpts is not included in WebdriverIO.Config typescript
**Is your feature request related to a problem? Please describe.**
WebDriverIO v5 cucumberOpts is not included in typescript . I have added wdio.d.ts file but it is nice to have adding in webdriverio package.

WebDriverIO version : 5.11.5

```
    interface Options {
        cucumberOpts?: {
            require: string[],
            backtrace: boolean,
            requireModule: (string | object)[],
            dryRun: boolean,
            failFast: boolean,
            format: string[],
            colors: boolean,
            snippets: boolean,
            source: boolean,
            profile: string[],
            strict: boolean,
            tagExpression: string,
            timeout: number,
            ignoreUndefinedDefinitions: boolean
        }
    }
```

### PR DESCRIPTION
**Pull request by jayrepo**
## Proposed changes
fix #4215
replace `tags` with `tagExpression` in wdio-cli template

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: scripts/templates/webdriverio.tpl.d.ts [hunk 1]
```diff
@@ -49,6 +49,24 @@ declare namespace WebdriverIO {
...
+        requireModule?: (string | object)[],
+        dryRun?: boolean,
+        failFast?: boolean,
+        format?: string[],
+        colors?: boolean,
+        snippets?: boolean,
+        source?: boolean,
+        profile?: string[],
+        strict?: boolean,
+        tags?: string[],
```
Comment (mgrybyk) [inline scripts/templates/webdriverio.tpl.d.ts]: Why do you removed tags from the template but have it here, can you make it consistent?

File: scripts/templates/webdriverio.tpl.d.ts [hunk 2]
```diff
@@ -49,6 +49,24 @@ declare namespace WebdriverIO {
         };
     }
 
+    interface CucumberOptions {
```
Comment (mgrybyk) [inline scripts/templates/webdriverio.tpl.d.ts]: Please move this one to cucumber type definition file

File: scripts/templates/webdriverio.tpl.d.ts [hunk 3]
```diff
@@ -66,6 +84,7 @@ declare namespace WebdriverIO {
         framework?: string,
         mochaOpts?: object,
         jasmineNodeOpts?: object,
+        cucumberOpts?: CucumberOptions,
```
Comment (mgrybyk) [inline scripts/templates/webdriverio.tpl.d.ts]: Same here, move to cucumber type definition

File: scripts/templates/webdriverio.tpl.d.ts (hunk 1 above)
Comment (jayrepo) [reply scripts/templates/webdriverio.tpl.d.ts]: Sorry, removed

File: scripts/templates/webdriverio.tpl.d.ts (hunk 2 above)
Comment (jayrepo) [reply scripts/templates/webdriverio.tpl.d.ts]: Moved to `cucumber-framework.d.ts`.
Will this throw error if user does not use `@wdio/cucumber-framework`?

File: scripts/templates/webdriverio.tpl.d.ts (hunk 3 above)
Comment (jayrepo) [reply scripts/templates/webdriverio.tpl.d.ts]: Same as above

File: scripts/templates/webdriverio.tpl.d.ts (hunk 2 above)
Comment (jayrepo) [reply scripts/templates/webdriverio.tpl.d.ts]: Shall I simply replace it with `object` like mochaOpts and jasmieOpts?

File: scripts/templates/webdriverio.tpl.d.ts [hunk 4]
```diff
@@ -66,6 +66,7 @@ declare namespace WebdriverIO {
         framework?: string,
         mochaOpts?: object,
         jasmineNodeOpts?: object,
+        cucumberOpts?: CucumberOpts,
```
Comment (mgrybyk) [inline scripts/templates/webdriverio.tpl.d.ts]: Remove it from here

File: packages/wdio-cucumber-framework/cucumber-framework.d.ts [hunk 5]
```diff
@@ -2,10 +2,12 @@ import { SourceLocation, ScenarioResult } from "cucumber";
 
 declare module "webdriverio" {
     interface HookFunctions extends CucumberHookFunctions {}
+    interface CucumberOpts extends CucumberOptions {}
```
Comment (mgrybyk) [inline packages/wdio-cucumber-framework/cucumber-framework.d.ts]: It should be just 
```
interface Options {
  cucumberOpts?: CucumberOptions;
}
```
Or 
`interface Options extends CucumberOptions {`

**Comment by mgrybyk** [comment]
Thank you!

**Comment by jayrepo** [comment]
@mgrybyk Thanks for your guidance on these typing stuff

**Pull request merged**